node/impl/full/eth.go

```diff
@@ -296,11 +301,41 @@ func (a *EthModule) EthGetBlockTransactionCountByHash(ctx context.Context, blkHa
 }
 
 func (a *EthModule) EthGetBlockByHash(ctx context.Context, blkHash ethtypes.EthHash, fullTxInfo bool) (ethtypes.EthBlock, error) {
-	ts, err := a.Chain.GetTipSetByCid(ctx, blkHash.ToCid())
+	cache := a.EthBlkCache
+	if fullTxInfo {
+		cache = a.EthBlkTxCache
+	}
+
+	// Attempt to retrieve the Ethereum block from cache
+	cid := blkHash.ToCid()
+	if cache != nil {
+		if ethBlock, found := cache.Get(cid); found {
+			if ethBlock != nil {
+				return *ethBlock, nil
+			}
+			// Log and remove the nil entry from cache
+			log.Errorw("nil value in eth block cache", "hash", blkHash.String())
+			cache.Remove(cid)
+		}
+	}
+
+	// Fetch the tipset using the block hash
+	ts, err := a.Chain.GetTipSetByCid(ctx, cid)
 	if err != nil {
-		return ethtypes.EthBlock{}, xerrors.Errorf("error loading tipset %s: %w", ts, err)
+		return ethtypes.EthBlock{}, xerrors.Errorf("failed to load tipset by CID %s: %w", blkHash.String(), err)
```

Comment:
xerrors.Errorf("failed to load tipset by CID %s: %w", cid.String(), err)

Reply:
Also, there's no need to explicitly call `String()`.